\documentclass[]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
  \newcommand{\euro}{â‚¬}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\usepackage{hyperref}
\PassOptionsToPackage{usenames,dvipsnames}{color} % color is loaded by hyperref
\hypersetup{unicode=true,
            pdftitle={Integrated Gene Set Analysis for microRNA Studies},
            pdfauthor={Analyzing just expressed genes},
            colorlinks=true,
            linkcolor=Maroon,
            citecolor=Blue,
            urlcolor=Blue,
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}

\title{Integrated Gene Set Analysis for microRNA Studies}
\author{Analyzing just expressed genes}
\date{2016-05-17}
\usepackage{amsmath}

% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

\begin{document}
\maketitle

\section{Introduction}\label{introduction}

In our paper we present methods and software for the functional
interpretation of microRNA expression data on its own. But many times,
extra information such as gene expression levels may be also available.
In such cases, researchers may want to restrict their functional
interpretation just to those target genes which are effectively
expressed. This sounds sensible as mRNA bridges miRNA functionality.

In this supplementary document we show how such analysis can be
trivially done using our \texttt{mdgsa} library.

\section{Methods}\label{methods}

When a list of expressed genes is available we can modify equation (2)
of the main paper as follows:

\begin{equation*}
t_i =
\begin{cases}
  \sum_{j \in G_i} r_{j}  & \text{if gene } i \text{ is expressed}\\
  0                       & \text{if gene } i \text{ is not expressed}\\
\end{cases}
\end{equation*}

This modification in the formula does not change our approach but
accounts for the non expression of the miRNA target genes.

\subsection{Code}\label{code}

In this section we provide the commands from the \texttt{mdgsa} library
used to carry out the functional interpretation of the microRNA
expression data. We indicate how they should be modified to exclude non
expressed genes as indicated in the above modification of equation (2).

Let \(p\) and \(s\) indicate respectively the \emph{p-values} and the
\emph{test statistics} resulting form the differential expression
analysis of the miRNA dataset. These will be R vectors.

Firs we will summarize the two quantities as indicated in
\textbf{equation (1)}. This can be done using the function
\texttt{pval2index} in the library:

\begin{verbatim}
index.mirna.level <- pval2index (pval = p, sign = t)
\end{verbatim}

Then we will \emph{transfer} the index from miRNA to mRNA level as
indicated in \textbf{equation (2)}. This can be done using the function
\texttt{transferIndex}:

\begin{verbatim}
index.gene.level <- transferIndex (index.mirna.level, targets)
\end{verbatim}

in this steps \texttt{targets} is an R \emph{list} containing gene
targets for each microRNA.

Up to here this is the standard implementation of our methodology. Now,
if we have a list of expressed genes in an R vector \texttt{EG}, we can
set to 0 the values in \texttt{index.gene.level} corresponding to genes
outside \texttt{EG} using the following command:

\begin{verbatim}
index.gene.level <- index.gene.level * names (index.gene.level) %in% EG
\end{verbatim}

This last trivial step will effectively transform equation (2) as
indicated above, setting to 0 all \emph{transferred} miRNA effects in
non expressed genes.

Then, the final step of the \emph{gene set analysis} can be carried out
as proposed in \textbf{equation (3)} using the function \texttt{uvGsa}:

\begin{verbatim}
res <- uvGsa (index.gene.level, annot)
\end{verbatim}

where \texttt{annot} is a list containing the gene annotation.

\bigskip

Notice that just the third command

\begin{verbatim}
index.gene.level <- index.gene.level * names (index.gene.level) %in% EG
\end{verbatim}

is needed to account for the non expressed genes. The standard analysis
when no information of the gene expression is available will still use
the same three remaining steps as devised in the library:

\begin{verbatim}
index.mirna.level <- pval2index (pval = p, sign = t)
index.gene.level <- transferIndex (index.mirna.level, targets)
res <- uvGsa (index.gene.level, annot)
\end{verbatim}

\section{Results}\label{results}

As a usage example, the above described approach was taken to reanalyze
the \textbf{KICH} (Kidney Chromophobe) dataset in TCGA.

Normalized Gene expression levels where downloaded from the TCGA web
site. Genes where considered expressed if they had normalized counts
above 1 for all of the samples in the study. With this, 8821 (73\%)
genes of the 12084 targeted by miRNAs where described as expressed. The
functional interpretation was carried out as described in the methods
section. All scripts used for the analysis can be found in:\\
\url{https://github.com/dmontaner-papers/gsa4mirna/tree/gh-pages/scripts_expression}

\bigskip

Overall, correlation between results of the functional interpretation
with and without the gene expression consideration was 0.5 for the
\emph{paired} analysis and 0.41 for the \emph{unpaired} one. Both
significant with a 0.05 confidence level.

Few GO terms such as \emph{homophilic cell adhesion via plasma membrane
adhesion molecules} (GO:0007156) and \emph{histone modification}
(GO:0016570) resulted significant in both analyses, while many others
where significant just when considering or not the expressed genes. This
may be a good indicator of the fundamental differences between the two
approaches.

All results of the analysis considering the expressed genes are
available in:\\
\url{https://github.com/dmontaner-papers/gsa4mirna/tree/gh-pages/supplementary_files_expression}

\end{document}
