<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <meta name="author" content="Analyzing just expressed genes" />
  <meta name="date" content="2016-05-17" />
  <title>Integrated Gene Set Analysis for microRNA Studies</title>
  <style type="text/css">code{white-space: pre;}</style>
  
</head>
<body>
<div id="header">
<h1 class="title">Integrated Gene Set Analysis for microRNA Studies</h1>
<h2 class="author">Analyzing just expressed genes</h2>
<h3 class="date">2016-05-17</h3>
</div>
<h1 id="introduction">Introduction</h1>
<p>In our paper we present methods and software for the functional interpretation of microRNA expression data on its own. But many times, extra information such as gene expression levels may be also available. In such cases, researchers may want to restrict their functional interpretation just to those target genes which are effectively expressed. This sounds sensible as mRNA bridges miRNA functionality.</p>
<p>In this supplementary document we show how such analysis can be trivially done using our <code>mdgsa</code> library.</p>
<h1 id="methods">Methods</h1>
<p>When a list of expressed genes is available we can modify equation (2) of the main paper as follows:</p>

<p>This modification in the formula does not change our approach but accounts for the non expression of the miRNA target genes.</p>
<h2 id="code">Code</h2>
<p>In this section we provide the commands from the <code>mdgsa</code> library used to carry out the functional interpretation of the microRNA expression data. We indicate how they should be modified to exclude non expressed genes as indicated in the above modification of equation (2).</p>
<p>Let <span class="math inline"><em>p</em></span> and <span class="math inline"><em>s</em></span> indicate respectively the <em>p-values</em> and the <em>test statistics</em> resulting form the differential expression analysis of the miRNA dataset. These will be R vectors.</p>
<p>Firs we will summarize the two quantities as indicated in <strong>equation (1)</strong>. This can be done using the function <code>pval2index</code> in the library:</p>
<pre><code>index.mirna.level &lt;- pval2index (pval = p, sign = t)</code></pre>
<p>Then we will <em>transfer</em> the index from miRNA to mRNA level as indicated in <strong>equation (2)</strong>. This can be done using the function <code>transferIndex</code>:</p>
<pre><code>index.gene.level &lt;- transferIndex (index.mirna.level, targets)</code></pre>
<p>in this steps <code>targets</code> is an R <em>list</em> containing gene targets for each microRNA.</p>
<p>Up to here this is the standard implementation of our methodology. Now, if we have a list of expressed genes in an R vector <code>EG</code>, we can set to 0 the values in <code>index.gene.level</code> corresponding to genes outside <code>EG</code> using the following command:</p>
<pre><code>index.gene.level &lt;- index.gene.level * names (index.gene.level) %in% EG</code></pre>
<p>This last trivial step will effectively transform equation (2) as indicated above, setting to 0 all <em>transferred</em> miRNA effects in non expressed genes.</p>
<p>Then, the final step of the <em>gene set analysis</em> can be carried out as proposed in <strong>equation (3)</strong> using the function <code>uvGsa</code>:</p>
<pre><code>res &lt;- uvGsa (index.gene.level, annot)</code></pre>
<p>where <code>annot</code> is a list containing the gene annotation.</p>
<p></p>
<p>Notice that just the third command</p>
<pre><code>index.gene.level &lt;- index.gene.level * names (index.gene.level) %in% EG</code></pre>
<p>is needed to account for the non expressed genes. The standard analysis when no information of the gene expression is available will still use the same three remaining steps as devised in the library:</p>
<pre><code>index.mirna.level &lt;- pval2index (pval = p, sign = t)
index.gene.level &lt;- transferIndex (index.mirna.level, targets)
res &lt;- uvGsa (index.gene.level, annot)</code></pre>
<h1 id="results">Results</h1>
<p>As a usage example, the above described approach was taken to reanalyze the <strong>KICH</strong> (Kidney Chromophobe) dataset in TCGA.</p>
<p>Normalized Gene expression levels where downloaded from the TCGA web site. Genes where considered expressed if they had normalized counts above 1 for all of the samples in the study. With this, 8821 (73%) genes of the 12084 targeted by miRNAs where described as expressed. The functional interpretation was carried out as described in the methods section. All scripts used for the analysis can be found in:<br />
<a href="https://github.com/dmontaner-papers/gsa4mirna/tree/gh-pages/scripts_expression" class="uri">https://github.com/dmontaner-papers/gsa4mirna/tree/gh-pages/scripts_expression</a></p>
<p></p>
<p>Overall, correlation between results of the functional interpretation with and without the gene expression consideration was 0.5 for the <em>paired</em> analysis and 0.41 for the <em>unpaired</em> one. Both significant with a 0.05 confidence level.</p>
<p>Few GO terms such as <em>homophilic cell adhesion via plasma membrane adhesion molecules</em> (GO:0007156) and <em>histone modification</em> (GO:0016570) resulted significant in both analyses, while many others where significant just when considering or not the expressed genes. This may be a good indicator of the fundamental differences between the two approaches.</p>
<p>All results of the analysis considering the expressed genes are available in:<br />
<a href="https://github.com/dmontaner-papers/gsa4mirna/tree/gh-pages/supplementary_files_expression" class="uri">https://github.com/dmontaner-papers/gsa4mirna/tree/gh-pages/supplementary_files_expression</a></p>
</body>
</html>
